security/pkg/pki/ca/ca.go

```diff
@@ -415,14 +417,37 @@ func (ca *IstioCA) sign(csrPEM []byte, subjectIDs []string, requestedLifetime ti
 }
 
 func (ca *IstioCA) signWithCertChain(csrPEM []byte, subjectIDs []string, requestedLifetime time.Duration, lifetimeCheck,
+	forCA bool) ([]string, error) {
+	cert, err := ca.sign(csrPEM, subjectIDs, requestedLifetime, lifetimeCheck, forCA)
+	if err != nil {
+		return nil, err
+	}
+	respCertChain := []string{string(cert)}
+	chainPem := ca.GetCAKeyCertBundle().GetCertChainPem()
+	if len(chainPem) > 0 {
+		respCertChain = append(respCertChain, string(chainPem))
+	}
+	return respCertChain, nil
+}
+
+func (ca *IstioCA) signWithCertChainWithBytesResult(csrPEM []byte, subjectIDs []string, requestedLifetime time.Duration, lifetimeCheck,
 	forCA bool) ([]byte, error) {
 	cert, err := ca.sign(csrPEM, subjectIDs, requestedLifetime, lifetimeCheck, forCA)
 	if err != nil {
 		return nil, err
 	}
+
 	chainPem := ca.GetCAKeyCertBundle().GetCertChainPem()
 	if len(chainPem) > 0 {
 		cert = append(cert, chainPem...)
 	}
 	return cert, nil
 }
+
+func (ca *IstioCA) GetCACertificatesFromMeshConfig() []*meshconfig.MeshConfig_CertificateData {
```

Comment:
See earlier comment - don't think there is a need to implement this function in the struct, nor alter the existing interface definition to include it

Reply:
yep, Thanks, removed from ca path